classes/install/Installer.inc.php

```diff
@@ -461,8 +461,10 @@ public function executeAction($action)
                     // Back out already-executed migrations.
                     while ($previousMigration = array_pop($this->migrations)) {
                         try {
-                            $previousMigration->down();
-                        } catch (PKP\install\DowngradeNotSupportedException $e) {
+                            if ($previousMigration->down() === true) {
```

Comment:
According to the method's signature, it returns boolean or null, but I don't see that any of those implementations return anything. Am I missing something?

Reply:
The only one that returns anything is the PreflightCheckMigration. None of the other migrations need to break out of a downgrade at the moment.

Reply:
But I had been assuming that a `?bool` return typehint would allow missing return statements without errors/warnings (taking a `void` as a `null`) but trying with another PHP release it looks like I was wrong :( I will re-work this.

Reply:
Done -- I didn't want to force all migrations to `return false` for a single migration's convenience, so I went instead with emptying out the installer's migration list in the preflight check when a fallback version is set and further downgrades aren't wanted. It's a much cleaner solution.